src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py

```diff
@@ -4,33 +4,47 @@
 # --------------------------------------------------------------------------------------------
 
 # pylint: disable=line-too-long
-from ._validators import get_datetime_type
+# pylint: disable=too-many-statements
+from azure.cli.core.commands.parameters import get_enum_type
+from ._validators import (datetime_type,
+                          decimal_type)
 
 
 def load_arguments(self, _):
     with self.argument_context('consumption usage') as c:
         c.argument('top', options_list=['--top', '-t'], type=int, help='Maximum number of items to return. Value range: 1-1000.')
         c.argument('include_additional_properties', options_list=['--include-additional-properties', '-a'], action='store_true', help='Include additional properties in the usages.')
         c.argument('include_meter_details', options_list=['--include-meter-details', '-m'], action='store_true', help='Include meter details in the usages.')
-        c.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='Start date (YYYY-MM-DD in UTC). If specified, also requires --end-date.')
-        c.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='End date (YYYY-MM-DD in UTC). If specified, also requires --start-date.')
+        c.argument('start_date', options_list=['--start-date', '-s'], type=datetime_type, help='Start date (YYYY-MM-DD in UTC). If specified, also requires --end-date.')
```

Comment:
Consider using the `get_datetime_type` method from `azure.cli.core.commands.parameters`. They are methods because you have to parameterize, but the intention will be to standardize date handling using these configurable methods.  Try it on one command and let me know whether it fits your scenario.